### PR TITLE
Validate user store used for user management.

### PR DESCRIPTION
**Pull request by sachithKay**

### CODE REVIEW

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java [hunk 1]
```diff
@@ -180,4 +188,21 @@ private boolean isAdmin(String[] rolesList) throws UserStoreException {
         return Arrays.asList(rolesList).contains(getRealmConfiguration().getAdminRoleName());
     }
 
+    protected Boolean isValidUserStore() {
+
+        Object fileUserStore = ConfigParser.getParsedConfigs().get("internal_apis.file_user_store.enable");
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java:193]: Can we use https://github.com/wso2/micro-integrator/blob/b6f301980b33d742918a7683b4e62bab6cbeb280/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityHandlerAdapter.java#L44 instead? then it'll be consistent

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java [hunk 2]
```diff
@@ -180,4 +188,21 @@ private boolean isAdmin(String[] rolesList) throws UserStoreException {
...
+        if (fileUserStore != null) {
+            if (!Boolean.valueOf(fileUserStore.toString())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected void setInvalidUserStoreResponse(org.apache.axis2.context.MessageContext axis2MessageContext) {
+        JSONObject response = Utils.createJsonError("User management is not supported with the in-memory user store. " +
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java:203]:
```suggestion
        JSONObject response = Utils.createJsonError("User management is not supported with the file-based user store. " +
```

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java (hunk 1 above)
Comment (sachithKay) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java]: changed as discussed

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityHandlerAdapter.java [hunk 3]
```diff
@@ -70,15 +71,21 @@ protected SecurityHandlerAdapter() {
...
+                    LOG.error("User store config has not been defined in file "
+                            + ManagementApiParser.getConfigurationFilePath(), e);
+                    isInitialized = false;
+                }
+            } else {
+                LOG.info("File based user store has been disabled. Carbon user store settings will be used.");
                 useCarbonUserStore = true;
-                LOG.info("User store config has not been defined in file "
-                         + ManagementApiParser.getConfigurationFilePath() + ". Using carbon user store settings");
+                isInitialized= true;
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityHandlerAdapter.java:87]:
```suggestion
                isInitialized = true;
```

**Comment by sachithKay** [comment]
Refactoring to remove the unused imports will be added with a separate PR.

**Pull request merged**